test-data/unit/check-isinstance.test

```diff
@@ -2377,24 +2378,24 @@ class B:
         ...
 
     def t0(self) -> None:
-        if isinstance(self, A0):        # E: Subclass of "B" and "A0" cannot exist: would have incompatible method signatures
+        if isinstance(self, A0):        # E: Subclass of "B & A0" cannot exist: would have incompatible method signatures
             x0: Literal[0] = self.f()   # E: Statement is unreachable
 
     def t1(self) -> None:
         if isinstance(self, A1):
-            reveal_type(self)           # N: Revealed type is "__main__.<subclass of "A1" and "B">"
-            x0: Literal[0] = self.f()   # E: Incompatible types in assignment (expression has type "Literal[1]", variable has type "Literal[0]")
-            x1: Literal[1] = self.f()
+            reveal_type(self)           # N: Revealed type is "__main__.B & __main__.A1"
+            x0: Literal[0] = self.f()   # E: Incompatible types in assignment (expression has type "Literal[1, 2]", variable has type "Literal[0]")
+            x1: Literal[1] = self.f()   # E: Incompatible types in assignment (expression has type "Literal[1, 2]", variable has type "Literal[1]")
```

Comment:
so previously this didnt work correctly?

Reply:
Previously it would create an intersection with the order based alphabetically, in that case this wasn't an error because it was selecting the member from the first entry of the intersection, now it creates the intersection based on narrowing order, eg:

```py
x: B
assert isinstance(x, A)
# x is "B & A"
```